odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,165 @@
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """ The forward-backward primal-dual splitting algorithm
+
+    Minimizes the sum of several convex functions composed with linear
+    operators
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``
+
+    Where f, g_i, h are convex functions, L_i are linear `Operator`'s, and h is
+    also differentiable.
+
+    Can also be used to solve the more general problem
+
+        ``min_x f(x) + sum_i (g_i @ l_i)(L_i x)``
+
+    where l_i are convex functions and @ is the infimal convolution:
+
+        ``(f @ g)(x) = inf_y { f(x-y) + g(y) }``
+
+    Parameters
+    ----------
+    x : `Vector`
+        Initial point
+    prox_f : `callable`
+        Funciton returning an `Operator` when called with `tau`.
+        The Operator should be the proximal of `tau * f`.
+    prox_cc_g : `sequence` of `callable`'s
+        Sequence of functions returning an operator when called with stepsize
+        `sigma`.
+        The Operator should be the proximal of `sigma_i * g_i^*`.
+    grad_h : `Operator`
+        Operators representing the gradient of  `h`.
+    L : `sequence` of `Operator`'s
+        A sequence with as many elements as ``prox__cc_gs`` of operators
+        ``L_i``
+    tau : `float`
+        Stepsize of ``f``
+    sigma : `sequence` of  `float`
+        Stepsize of the ``g_i``'s
+    niter : `int`
+        Number of iterations
+    callback : `Callback`, optional
+        Show partial results
+
+    Other Parameters
+    ----------------
+    grad_cc_l : `sequence` of `Operator`'s, optional
+        Sequence of operators representing the gradient of  `l_i^*`.
+        If omitted, the simpler problem will be considered, which corresponds
+        to the convex conjugate functionals of the l_i:s being zero functionals
+        and hence the gradient being the zero-operator.
+
+    Notes
+    -----
+    Strictly, we have the following conditions on the functions involved: f and
+    g_i are proper, convex and lower semicontinuous, and h is convex and
+    differentialbe with :math:`\\eta`-Lipschitz continuous gradient.
+
+    For the optional input l_i we need it to be proper, convex, lower
+    semicontinuous, and :math:`\\nu_i^{-1}`-strongly convex. The fact that l_i
+    is :math:`\\nu_i^{-1}`-strongly convex implies that the convex conjugate
+    functional of l_i is differentiable with :math:`\\nu_i`-Lipschitz
+    continuous gradient.
+
+    To guarantee convergence, the parameters ``tau``, ``sigma`` and
+    ``L`` need to satisfy
+
+    .. math::
+
+       2 * \min \{ \\frac{1}{\\tau}, \\frac{1}{\sigma_1}, \\ldots,
+       \\frac{1}{\sigma_m} \} * \min\{ \\eta, \\nu_1, \\ldots, \\nu_m  \} *
+       \\sqrt{1 - \\tau \\sum_{i=1}^n \\sigma_i ||L_i||^2} > 1
+
+    where, if the simpler problem is considered, all :math:`\\nu_i` can be
+    considered to be :math:`\\infty`.
```

Comment:
Some guideline of parameter choice would be great, i.e. is there some easy way of getting the condition right? Something like "choose one parameter such and such, then the other one can be picked such and such".